### PR TITLE
sync swagger with upstream

### PR DESCRIPTION
**Pull request by bkleiner**
*Description of changes:*
recent changes in firecracker made the sdk incompatible with the firecracker api.
this pull-request syncs the swagger.yaml and integrates the changes with the rest of the codebase.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: client/swagger.yaml [hunk 1]
```diff
@@ -395,7 +385,7 @@ definitions:
 
   Error:
     properties:
-      fault_message:
+      faultMessage:
```
Comment (nmeyerhans) [inline client/swagger.yaml]: We actually don't want this change. `fault_message` is correct, and there's an open PR to get it fixed in firecracker. See https://github.com/firecracker-microvm/firecracker/pull/713#discussion_r238536099

**Comment by samuelkarp** [comment]
@bkleiner It looks like the `fault_message` change was reverted in https://github.com/firecracker-microvm/firecracker/pull/713.  Do you want to update this PR?

**Comment by bkleiner** [comment]
@samuelkarp Yep, just did not get around to it, the change is online now.

**Comment by samuelkarp** [comment]
I'm getting a panic when I try to test this:

```
panic: interface conversion: interface {} is *operations.PutMmdsDefault, not *operations.PutMmdsNoContent [recovered]
	panic: interface conversion: interface {} is *operations.PutMmdsDefault, not *operations.PutMmdsNoContent

goroutine 89 [running]:
testing.tRunner.func1(0xc000372900)
	/home/skarp/.gobin/go/src/testing/testing.go:792 +0x387
panic(0x875160, 0xc0004b2e10)
	/home/skarp/.gobin/go/src/runtime/panic.go:513 +0x1b9
github.com/firecracker-microvm/firecracker-go-sdk/client/operations.(*Client).PutMmds(0xc00000df40, 0xc000431380, 0x30, 0xc000431350, 0xc0004c1ea0)
	/home/skarp/projects/firecracker-go-sdk/client/operations/operations_client.go:150 +0x44a
github.com/firecracker-microvm/firecracker-go-sdk.(*FirecrackerClient).PutMmds(0xc0000a5158, 0x96ed60, 0xc0000a2010, 0x864440, 0xc000431350, 0x8ab040, 0xc000132c60, 0x4712aa8dc5726f52)
	/home/skarp/projects/firecracker-go-sdk/firecracker.go:140 +0x8f
github.com/firecracker-microvm/firecracker-go-sdk.(*Machine).SetMetadata(0xc0000b04e0, 0x96ed60, 0xc0000a2010, 0x864440, 0xc000431350, 0x36, 0x44d1a8)
	/home/skarp/projects/firecracker-go-sdk/machine.go:512 +0x83
github.com/firecracker-microvm/firecracker-go-sdk.testSetMetadata(0x96ed60, 0xc0000a2010, 0xc000372900, 0xc0000b04e0)
	/home/skarp/projects/firecracker-go-sdk/machine_test.go:360 +0xb5
github.com/firecracker-microvm/firecracker-go-sdk.TestMicroVMExecution.func10(0xc000372900)
	/home/skarp/projects/firecracker-go-sdk/machine_test.go:134 +0x46
testing.tRunner(0xc000372900, 0xc0003b5340)
	/home/skarp/.gobin/go/src/testing/testing.go:827 +0xbf
created by testing.(*T).Run
	/home/skarp/.gobin/go/src/testing/testing.go:878 +0x353
exit status 2
FAIL	github.com/firecracker-microvm/firecracker-go-sdk	2.045s
```

**Comment by samuelkarp** [comment]
This seems to work against the `master` branch of Firecracker but not against the v0.11.0 tag.  I think we're going to wait on merging this until there is a tagged release with these changes in it.

**Comment by bkleiner** [comment]
https://github.com/firecracker-microvm/firecracker/pull/771 is under its way. it would be awesome if we could merge once that one is done.

**Comment by nmeyerhans** [comment]
The new release is 0.12.0 and includes a couple additional changes that aren't captured here. I'm going to close this and open a new PR to update the SDK for 0.12.0 momentarily.

**Pull request closed without merging**